Ruby 2.2.3 will not build on i386 Tiger

https://gist.github.com/14fa08329609ac8132538c054bec33f7

Relevant part, from signal.c:

```
compiling signal.c
signal.c: In function ‘check_stack_overflow’:
signal.c:778: error: ‘struct mcontext’ has no member named ‘__ss’
make: *** [signal.o] Error 1
```

The code in question:

```c
# elif defined __APPLE__
#   if defined(__LP64__)
    const uintptr_t sp = mctx->__ss.__rsp;
#   else
    const uintptr_t sp = mctx->__ss.__esp;
#   endif
```

--- a/Library/Formula/ruby.rb
+++ b/Library/Formula/ruby.rb
@@ -35,6 +35,27 @@ class Ruby < Formula
   end
 
   def install
+    # mcontext types had a member named `ss` instead of `__ss`
+    # prior to Leopard; see
+    # https://github.com/mistydemeo/tigerbrew/issues/473
+    if Hardware::CPU.intel? && MacOS.version < :leopard
+      inreplace "signal.c" do |s|
+        s.gsub! "->__ss.", "->ss."
+        s.gsub! "__rsp", "rsp"
+        s.gsub! "__esp", "esp"
+      end
+
+      inreplace "vm_dump.c" do |s|
+        s.gsub! /uc_mcontext->__(ss)\.__(r\w\w)/,
+                "uc_mcontext->\1.\2"
+        s.gsub! "mctx->__ss.__##reg",
+                "mctx->ss.reg"
+        # missing include in vm_dump; this is an ugly solution
+        s.gsub! '#include "iseq.h"',
+                %{#include "iseq.h"\n#include <ucontext.h>}
+      end
+    end
+
     system "autoconf" if build.head?
 
     args = %W[

